WcaOnRails/app/views/competitions/_index_table.html.erb

```diff
@@ -17,7 +17,7 @@
         <% else %>
           <%= icon('hourglass-start', title: "Starts in #{ pluralize((competition.start_date - Date.today).to_i, "day") }", data: { toggle: "tooltip" }) %>
         <% end %>
-        <%= wca_date_range(competition.start_date, competition.end_date)  %>
+        <%= competition.start_date && competition.end_date ? wca_date_range(competition.start_date, competition.end_date) : "Unscheduled" %>
```

Comment:
Oh yikes. We have a number of additional validations for confirming a competition (including requiring that [dates be present](https://github.com/jfly/worldcubeassociation.org/blob/9998edd118e34b0d86df316ebb2a58ffab6f64fe/WcaOnRails/app/models/competition.rb#L396-L400)), but there's probably nothing that would prevent a board member from making a competition and then making it visible without first confirming it. We should enforce this "extra validations" both when confirming and when making a competition visible to the public.

@pingskills, in the interest of not repeating ourselves, I've changed my mind and decided I'd be ok with moving the "Unscheduled" out of the views and into the `wca_date_range` method. In my previous job, we had a very strong culture of writing code that would fail-fast, rather than limping along in some scenario the programmer didn't intend for. My knee jerk reaction is to apply that philosophy to everything, but I think the conciseness and DRY-ness of moving the "Unscheduled" string into the `wca_date_range` method is worth it here. Also, it would be *really* awesome if you pulled the "Unscheduled" string out into a yml file, as per @viroulep's recent work to start getting the website translated (see https://github.com/cubing/worldcubeassociation.org/pull/460).